### PR TITLE
Filters media by media types

### PR DESCRIPTION
**Pull request by Daniel-KM**
Hi, 
Often, scanned manuscripts have not only a list of images, but a pdf too, because some people prefer to download them. There may be some other type of files too that don't need to be managed by Scripto. 

So this pr add a field to limit the medias to a list of media types, if wanted. So the user sees only the jpeg to transcribe, not the pdf, nor other attached files. This fix is already working on a living site.

I can divide the pr if needed, I added some fixes for translation.

### CODE REVIEW

File: src/Job/SyncProject.php [hunk 1]
```diff
@@ -136,14 +137,24 @@ public function syncProjectMedia(ScriptoProject $project)
...
             // abstraction ever changes.
-            $query = $em->createQuery('
-                DELETE FROM Scripto\Entity\ScriptoMedia sm
-                WHERE sm.scriptoItem = :scripto_item_id
-                AND sm.id NOT IN (:scripto_media_ids)
-            ')->setParameters([
-                'scripto_item_id' => $sItemId,
-                'scripto_media_ids' => $sMediaIdsToRetain,
-            ]);
+            if ($sMediaIdsToRetain) {
```
Comment (jimsafley) [inline src/Job/SyncProject.php:140]: What problem are you solving with this change?

Comment (Daniel-KM) [reply src/Job/SyncProject.php:140]: When there are no media ids to retain (the item has only a pdf, no jpeg, and the pdf is excluded), the query is not correct : `AND sm.id NOT IN ()`.

**Comment by jimsafley** [comment]
Apologies for the late response, but I'm not going to merge this PR because it contains changes beyond the stated "Filters media by media types" feature. Perhaps if you split the changes into separate PRs I can better evaluate them.

**Comment by jimsafley** [comment]
Thanks for this useful feature!

File: src/Job/SyncProject.php [hunk 2]
```diff
@@ -193,10 +204,19 @@ public function getScriptoMediaEntity(ScriptoProject $project, Item $item, Media
...
      */
     public function getAllItemMedia(Item $item, ScriptoProject $project)
     {
-        return $item->getMedia();
+        $medias = $item->getMedia();
+        $mediaTypes = $project->getMediaTypes();
+        if ($mediaTypes) {
+            $mediaTypes[] = '';
+            $criteria = \Doctrine\Common\Collections\Criteria::create()
+                ->where(Criteria::expr()->isNull('mediaType'))
```
Comment (jimsafley) [inline src/Job/SyncProject.php:216]: @Daniel-KM We noticed this peculiarity during testing. We're pretty sure a more expected behavior is to _exclude_ media with NULL media type if the project owner explicitly selects media types. Is there a reason you're _including_ them here?

Comment (Daniel-KM) [reply src/Job/SyncProject.php:216]: It's more related to the database; i saw an error when this criteria is not applied.

Comment (jimsafley) [reply src/Job/SyncProject.php:216]: An error there is alarming. I did not see an error after removing that line. Under what conditions did you see the error? We strongly prefer to exclude NULL media types, so we need to resolve this before we make a release.

Comment (Daniel-KM) [reply src/Job/SyncProject.php:216]: I don't remember, so it can be skipped.

**Pull request merged**